nemo/collections/nlp/modules/common/megatron/position_embedding/rotary_position_embedding.py

```diff
@@ -81,5 +178,8 @@ def apply_rotary_pos_emb(t, freqs):
     t, t_pass = t[..., :rot_dim], t[..., rot_dim:]
     # first part is cosine component
     # second part is sine component, need to change signs with _rotate_half method
-    t = (t * freqs.cos()) + (_rotate_half(t) * freqs.sin())
+    if mscale:
+        t = (t * freqs.cos() * mscale) + (_rotate_half(t) * freqs.sin() * mscale)
+    else:
+        t = (t * freqs.cos()) + (_rotate_half(t) * freqs.sin())
     return torch.cat((t, t_pass), dim=-1)
```

Comment:
I think maybe we should multiply mscale to all `torch.cat((t, t_pass), dim=-1)` since the original yarn has rope for all dimension but we only have 50% dimension in our models.